src/bigquery.ts

```diff
@@ -0,0 +1,1177 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ** This file is automatically generated **
+// ** All changes to this file may be overwritten. **
+
+import {
+  protos,
+  DatasetServiceClient,
+  TableServiceClient,
+  JobServiceClient,
+  ModelServiceClient,
+  RoutineServiceClient,
+  RowAccessPolicyServiceClient,
+} from '.';
+import type * as gax from 'google-gax';
+import {
+  Callback,
+  CallOptions,
+  ClientOptions,
+  PaginationCallback,
+} from 'google-gax';
+/**
+ * Options passed to the underlying client.
+ *
+ * @param {object} [options] - The configuration object.
+ * The options accepted by the constructor are described in detail
+ * in [this document](https://github.com/googleapis/gax-nodejs/blob/main/client-libraries.md#creating-the-client-instance).
+ * The common options are:
+ * @param {object} [options.credentials] - Credentials object.
+ * @param {string} [options.credentials.client_email]
+ * @param {string} [options.credentials.private_key]
+ * @param {string} [options.email] - Account email address. Required when
+ *     using a .pem or .p12 keyFilename.
+ * @param {string} [options.keyFilename] - Full path to the a .json, .pem, or
+ *     .p12 key downloaded from the Google Developers Console. If you provide
+ *     a path to a JSON file, the projectId option below is not necessary.
+ *     NOTE: .pem and .p12 require you to specify options.email as well.
+ * @param {number} [options.port] - The port on which to connect to
+ *     the remote host.
+ * @param {string} [options.projectId] - The project ID from the Google
+ *     Developer's Console, e.g. 'grape-spaceship-123'. We will also check
+ *     the environment variable GCLOUD_PROJECT for your project ID. If your
+ *     app is running in an environment which supports
+ *     {@link https://developers.google.com/identity/protocols/application-default-credentials Application Default Credentials},
+ *     your project ID will be detected automatically.
+ * @param {string} [options.apiEndpoint] - The domain name of the
+ *     API remote host.
+ * @param {gax.ClientConfig} [options.clientConfig] - Client configuration override.
+ *     Follows the structure of {@link gapicConfig}.
+ * @param {boolean} [options.fallback] - Use HTTP/1.1 REST mode.
+ *     For more information, please check the
+ *     {@link https://github.com/googleapis/gax-nodejs/blob/main/client-libraries.md#http11-rest-api-mode documentation}.
+ * @param {gax} [gaxInstance]: loaded instance of `google-gax`. Useful if you
+ *     need to avoid loading the default gRPC version and want to use the fallback
+ *     HTTP implementation. Load only fallback version and pass it to the constructor:
+ *     ```
+ *     const gax = require('google-gax/build/src/fallback'); // avoids loading google-gax with gRPC
+ *     const client = new DatasetServiceClient({fallback: true}, gax);
+ *     ```
+ */
+export type subClientOptions = {
```

Comment:
nit: should those exported types start with uppercase ?

Reply:
yup! done.